File: backend_ambulance/AmbulanceManagement/src/main/java/pl/jakub/ambulancemanagement/shifts/model/Shift.java
package pl.jakub.ambulancemanagement.shifts.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import pl.jakub.ambulancemanagement.ambulances.model.Ambulance;
import pl.jakub.ambulancemanagement.users.model.User;

import java.time.LocalDateTime;

@Setter
@Getter
@NoArgsConstructor
@Entity
@Table(name = "shifts")

public class Shift {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "driver_id", nullable = false)
    private User driver;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ambulance_id", nullable = false)
    private Ambulance ambulance;

    @Enumerated(EnumType.STRING)
    @Column(name = "shift_type", nullable = false)
    private  ShiftType shiftType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_id", nullable = false)
    private User createdBy;

    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;

    @Column(name = "end_time", nullable = false)
    private LocalDateTime endTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ShiftStatus status;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: frontend_ambulance/frontend-ambulance/src/types/shift.ts
export type ShiftType = "DAY_12H" | "NIGHT_12H" | "FULL_24H" | "OTHER";

export type ShiftStatus = "ACTIVE" | "FINISHED" | "CANCELLED";

export type ShiftDefaultMemberCreateRequest = {
  userId: number;
  startTime: string;
  endTime: string | null;
};

export type ShiftCreateRequest = {
  ambulanceId: number;
  shiftType: ShiftType;
  startTime: string;
  endTime: string;
  defaultMember: ShiftDefaultMemberCreateRequest | null;
};

export type ShiftResponse = {
  id: number;
  driverId: number;
  ambulanceId: number;
  shiftType: ShiftType;
  createdById: number;
  startTime: string;
  endTime: string;
  status: ShiftStatus;
  createdAt: string;
};